server/task/src/main/scala/org/apache/james/task/eventsourcing/TaskAggregate.scala

```diff
@@ -30,54 +30,54 @@ import scala.jdk.CollectionConverters._
 
 class TaskAggregate private(val aggregateId: TaskAggregateId, private val history: History) {
 
-  private val decisionProjection: Option[DecisionProjection] = DecisionProjection.load(history)
-  private val currentStatus: Option[Status] = decisionProjection.map(_.status)
+  private val currentStatus: Option[Status] = DecisionProjection
+    .load(history.getEvents.asScala.toList)
+    .map(_.status)
 
   def create(task: Task): util.List[Event] = {
     if (currentStatus.isEmpty) {
-      publishEvents(history, Created(aggregateId, _, task))
+      createEventWithId(Created(aggregateId, _, task))
     } else Nil.asJava
   }
 
   private[eventsourcing] def start(): util.List[Event] = {
     currentStatus match {
-      case Some(Status.WAITING) => publishEvents(history, Started(aggregateId, _))
+      case Some(Status.WAITING) => createEventWithId(Started(aggregateId, _))
       case _ => Nil.asJava
     }
   }
 
   def requestCancel(): util.List[Event] = {
     currentStatus match {
-      case Some(status) if !status.isFinished => publishEvents(history, CancelRequested(aggregateId, _))
+      case Some(status) if !status.isFinished => createEventWithId(CancelRequested(aggregateId, _))
       case _ => Nil.asJava
     }
   }
 
   private[eventsourcing] def complete(result: Result): util.List[Event] = {
     currentStatus match {
-      case Some(status) if !status.isFinished => publishEvents(history, Completed(aggregateId, _, result))
+      case Some(status) if !status.isFinished => createEventWithId(Completed(aggregateId, _, result))
       case _ => Nil.asJava
     }
   }
 
   private[eventsourcing] def fail(): util.List[Event] = {
     currentStatus match {
-      case Some(status) if !status.isFinished => publishEvents(history, Failed(aggregateId, _))
+      case Some(status) if !status.isFinished => createEventWithId(Failed(aggregateId, _))
       case _ => Nil.asJava
     }
   }
 
   private[eventsourcing] def cancel(): util.List[Event] = {
     currentStatus match {
-      case Some(status) if !status.isFinished => publishEvents(history, Cancelled(aggregateId, _))
+      case Some(status) if !status.isFinished => createEventWithId(Cancelled(aggregateId, _))
       case _ => Nil.asJava
     }
   }
 
-  private def publishEvents(history: History, events: (EventId => Event)*): util.List[Event] =
-    Stream.iterate(history.getNextEventId)(_.next())
-      .zip(events)
-      .map({ case (eventId, builder) => builder(eventId) })
+  private def createEventWithId(events: EventId => Event): util.List[Event] =
```

Comment:
s/events/eventWithoutId/ ?

Reply:
I don't think its relevant. The parameter is a function constructing an event from an EventId.
But i agree with you that the name could be better. Do you have another idea ?

Reply:
is eventWithoutId better than events? If the answer is yes, use that, otherwise keep events.
BTW, there's a single event so at the very least you should drop the s.